### PR TITLE
Closes #9484: Handle multiple add-ons update that require new permissions

### LINKED ISSUE #9484: AddUpdater doesn't handle multiple notifications for simultaneously add-ons updates
We an add-on update requires new permissions, a notification is created for the user to allow or deny the new permissions. Our actual implementation, creates only one notification if multiple add-ons update require new permissions

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by Amejia481** [comment]
I added two commits:

* Fixing the Android 11 deprecation:  https://github.com/mozilla-mobile/android-components/pull/9489/commits/c7cc4a00bcdb41743939a7dae0559a82dab5030b

* Handle multiple add-ons update that require new permissions https://github.com/mozilla-mobile/android-components/pull/9489/commits/ca77b3c73aaa57f203cdf1bc4c57e2d3b0642024

For the last one, I forced to installed [this old version](https://addons.mozilla.org/firefox/downloads/file/641006/video_background_play_fix-1.0-fx.xpi) of Video background play fix, which has less permissions than the latest one, then I an forced an update to happen this way,  I was able to see the feature is still working as expected.

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt [hunk 1]
```diff
@@ -269,7 +270,7 @@ class DefaultAddonUpdater(
         extension: WebExtension,
         newPermissions: List<String>
     ): Notification {
-        val notificationId = SharedIdsHelper.getIdForTag(applicationContext, NOTIFICATION_TAG)
+        val notificationId = NotificationHandlerService.getNotificationId(applicationContext, extension.id)
```
Comment (Amejia481) [inline components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt:273]: The main issue that prevented multiple notifications from showing was that we were using the same notification id, now to create the notification id we are including the `extension.id` as well, making the notification unique per extensions.

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt [hunk 2]
```diff
@@ -408,42 +412,53 @@ class DefaultAddonUpdater(
...
+            onHandleIntent(intent)
+            return super.onStartCommand(intent, flags, startId)
+        }
+
+        override fun onBind(intent: Intent?): IBinder? = null
+
+        companion object {
+            @VisibleForTesting
+            internal fun getNotificationId(context: Context, addonId: String): Int {
+                return SharedIdsHelper.getIdForTag(context, NOTIFICATION_TAG + addonId)
```
Comment (csadilek) [inline components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt:460]: nit: do you want this to be + "." + addonId. Separated by a dot instead? 
e.g. `mozilla.components.feature.addons.update.addonUpdater.12345`

**Review by csadilek — APPROVED** [review]
👍

File: components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt (hunk 2 above)
Comment (Amejia481) [reply components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt:460]: Sure 👍🏽

**Pull request merged**